include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -83,6 +83,98 @@ namespace internal
 
 
 
+    /**
+     * Generate evalution points on a simplex with arbitrary number of
+     * subdivisions.
+     */
+    template <int dim>
+    inline std::vector<Point<dim>>
+    generate_simplex_evaluation_points(const unsigned int n_subdivisions)
+    {
+      (void)n_subdivisions;
+
+      Assert(false, ExcNotImplemented());
+
+      return {};
+    }
+
+
+
+    /**
+     * Helper function to create evaluation points recursivly.
+     */
+    inline void
+    generate_simplex_evaluation_points_recursivly(
+      const std::vector<Point<2>> &bounding_vertices,
+      const unsigned int           n_subdivisions,
+      std::vector<Point<2>> &      evalution_points)
+    {
+      if (n_subdivisions == 0)
+        {
+          evalution_points.push_back(bounding_vertices[0]);
+          return;
+        }
+
+      for (const auto &p : bounding_vertices)
+        evalution_points.push_back(p);
+
+      if (n_subdivisions == 1)
+        return;
+
+      const auto process = [&](const Point<2> &p0, const Point<2> &p1) {
+        std::vector<Point<2>> line_points;
+
+        for (unsigned int i = 1; i < n_subdivisions; ++i)
+          line_points.push_back(p0 + (p1 - p0) / n_subdivisions * i);
+
+        evalution_points.insert(evalution_points.end(),
+                                line_points.begin(),
+                                line_points.end());
+
+        return line_points;
+      };
+
+      const auto line_points_0 =
+        process(Point<2>(bounding_vertices[0]), Point<2>(bounding_vertices[1]));
+      const auto line_points_1 =
+        process(Point<2>(bounding_vertices[1]), Point<2>(bounding_vertices[2]));
+      const auto line_points_2 =
+        process(Point<2>(bounding_vertices[2]), Point<2>(bounding_vertices[0]));
+
+      if (n_subdivisions == 2)
+        return;
+
+      generate_simplex_evaluation_points_recursivly(
+        {{Point<2>(line_points_0[line_points_0.size() - 2][0],
+                   line_points_1[0][1]),
+          Point<2>(line_points_0[0][0],
+                   line_points_1[line_points_1.size() - 2][1]),
+          Point<2>(line_points_0[0][0], line_points_1[0][1])}},
+        n_subdivisions - 3,
+        evalution_points);
+    }
+
+
+
+    /**
+     * Specialization for triangles.
+     */
+    template <>
+    inline std::vector<Point<2>>
+    generate_simplex_evaluation_points(const unsigned int n_subdivisions)
+    {
+      std::vector<Point<2>> evalution_points;
+
+      generate_simplex_evaluation_points_recursivly(
+        {{Point<2>(0.0, 0.0), Point<2>(1.0, 0.0), Point<2>(0.0, 1.0)}},
+        n_subdivisions,
+        evalution_points);
+
+      return evalution_points;
+    }
```

Comment:
We need some kind of `QIterated` for arbitrary reference-cell types.  For now, I am position the evaluation points as depicted in the documentation of VTK (https://blog.kitware.com/wp-content/uploads/2018/09/Source_Issue_43.pdf). However, we might want a different ordering.

Reply:
What would be the difficulty in just generalizing `QIterated`? We'll need that one day anyway.